tests/testthat/test-whoami.R

```diff
@@ -0,0 +1,25 @@
+context("whoami")
+
+test_that("whoami works in presence of PAT", {
+  skip_on_travis()
+  skip_on_appveyor()
+  res <- gh_whoami()
+  res$token <- NULL
+  expect_equivalent(
+    res,
+    list(name = "Jennifer (Jenny) Bryan",
+         login = "jennybc",
+         html_url = "https://github.com/jennybc",
+         scopes = "admin:org, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user"))
+})
+
+test_that("whoami works in absence of PAT", {
+  expect_message(res <- gh_whoami(.token = ""),
```

Comment:
I think would be great if `expect_error` retained the error object, and we could test the error class. See https://github.com/hadley/testthat/issues/530